pyvista/core/utilities/reader.py

```diff
@@ -2085,6 +2087,168 @@ def set_active_time_point(self, time_point) -> None:  # noqa: D102
+        self.set_active_time_point(0)
+
+    @property
+    def number_time_points(self):
+        """Return number of time points or iterations available to read.
+
+        Returns
+        -------
+        int
+
+        """
+        return self.reader.GetNumberOfTimeSteps()
+
+    @property
+    def time_values(self):
+        """All time or iteration values.
+
+        Returns
+        -------
+        list[float]
+
+        """
+        vtkStreaming = _lazy_vtk_instantiation(
+            'vtkCommonExecutionModel', 'vtkStreamingDemandDrivenPipeline'
+        )
+        key = vtkStreaming.TIME_STEPS()
+
+        vtkinfo = self.reader.GetExecutive().GetOutputInformation(0)
+        return [vtkinfo.Get(key, i) for i in range(self.number_time_points)]
+
+    def time_point_value(self, time_point):
+        """Value of time point or iteration by index.
+
+        Parameters
+        ----------
+        time_point : int
+            Time point index.
+
+        Returns
+        -------
+        float
+
+        """
+        return self.time_values[time_point]
+
+    @property
+    def active_time_value(self):
+        """Active time or iteration value.
+
+        Returns
+        -------
+        float
+
+        """
+        vtkStreaming = _lazy_vtk_instantiation(
+            'vtkCommonExecutionModel', 'vtkStreamingDemandDrivenPipeline'
+        )
+        key = vtkStreaming.UPDATE_TIME_STEP()
+        vtkinfo = self.reader.GetExecutive().GetOutputInformation(0)
+        return vtkinfo.Get(key)
+
+    @property
+    def active_time_point(self):
+        """Active time point.
+
+        Returns
+        -------
+        int
+
+        """
+        return self.time_values.index(self.active_time_value)
+
+    def set_active_time_value(self, time_value):
+        """Set active time or iteration value.
+
+        Parameters
+        ----------
+        time_value : float
+            Time or iteration value to set as active.
+
+        """
+        vtkStreaming = _lazy_vtk_instantiation(
+            'vtkCommonExecutionModel', 'vtkStreamingDemandDrivenPipeline'
+        )
+        key = vtkStreaming.UPDATE_TIME_STEP()
+        vtkinfo = self.reader.GetExecutive().GetOutputInformation(0)
+        vtkinfo.Set(key, time_value)
+
+        self.reader.Update()
+
+    def set_active_time_point(self, time_point):
+        """Set active time or iteration by index.
+
+        Parameters
+        ----------
+        time_point : int
+            Time or iteration point index for setting active time.
+
+        """
+        if time_point < 0 or time_point >= self.number_time_points:
+            raise ValueError(f'Time point out of range [0, {self.number_time_points-1}]')
+
+        self.set_active_time_value(self.time_values[time_point])
+
+    @property
+    def number_cell_arrays(self):  # noqa: D102
+        raise AttributeError(
+            'Nek 5000 data does not contain cell arrays, ' 'this method cannot be used'
```

Comment:
This looks like a ruff reformatting issue?  Can it be a single string?  Applies to all below . It would be cleaner to have a private variable defined once for these strings, but not critical.

Reply:
I have made them private